libcaf_core/caf/node_id.hpp

```diff
@@ -288,7 +288,7 @@ namespace std {
 
 template <>
 struct hash<caf::node_id> {
-  size_t operator()(const caf::node_id& x) const noexcept {
```

Comment:
Also should stay noexcept.

Reply:
Since this is calling the above function, `fnv::compute` which can throw because of `f.apply`, we get a warning here about an exception being thrown.

Reply:
I'll return the noexcept and ignore clang-tidy, for the best.

Reply:
`fnv` should be fully noexcept. `clang-tidy` shouldn't complain in the first place. :)